python/cudf/cudf/core/buffer.py

```diff
@@ -171,62 +203,56 @@ def deserialize(cls, header: dict, frames: list) -> Buffer:
 
         return buf
 
-    @classmethod
-    def empty(cls, size: int) -> Buffer:
-        return Buffer(rmm.DeviceBuffer(size=size))
 
-    def copy(self) -> Buffer:
-        """
-        Create a new Buffer containing a copy of the data contained
-        in this Buffer.
-        """
-        from rmm._lib.device_buffer import copy_device_to_ptr
+def _get_ptr_and_size(array_interface: Mapping) -> Tuple[int, int]:
+    """
+    Return the pointer and size of an array interface.
 
-        out = Buffer.empty(size=self.size)
-        copy_device_to_ptr(self.ptr, out.ptr, self.size)
-        return out
+    Raises ValueError if array isn't c-contiguous
+    """
 
+    def is_c_contiguous(shape, strides, itemsize):
+        if strides is None or any(dim == 0 for dim in shape):
+            return True
+        sd = itemsize
```

Comment:
Maybe name `sd` something like `multi_stride`?

Reply:
Renamed to `cumulative_stride`